### PR TITLE
Update base-fi.yaml

### PR DESCRIPTION
**Pull request by turbomursu**
Full translation. Fixed some bad grammar. Anything you need here?

### CODE REVIEW

File: translations/base-fi.yaml [hunk 1]
```diff
@@ -662,59 +654,53 @@ storyRewards:
...
     reward_constant_signal:
-        title: Constant Signal
-        desc: You unlocked the <strong>constant signal</strong> building on the wires
-            layer! This is useful to connect it to <strong>item filters</strong>
-            for example.<br><br> The constant signal can emit a
-            <strong>shape</strong>, <strong>color</strong> or
-            <strong>boolean</strong> (1 / 0).
+        title: Jatkuva Signaali
+        desc: Avasit <strong>Jatkuvan Signaalin</strong> laitteen johtotasolla!
+		Tämä on hyödyllinen esimerkiksi yhdistettynä
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:693]: Too many spaces in front This needs to be 3 Tabs (12 spaces)

File: translations/base-fi.yaml [hunk 2]
```diff
@@ -662,59 +654,53 @@ storyRewards:
...
+		<strong>tavarasuodattimeen.</strong><br><br> Jatkuva signaali voi lähettää esim.
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:694]: Too many spaces in front This needs to be 3 Tabs (12 spaces)

File: translations/base-fi.yaml [hunk 3]
```diff
@@ -831,84 +817,72 @@ settings:
...
             description: Ottaa käyttöön vinjetoinnin, joka tummentaa näytön kulmia ja tekee
                 tekstin lukemisesta helpompaa.
         rotationByBuilding:
-            title: Kiertäminen rakennustyypin mukaan
-            description: Jokainen rakennustyyppi muistaa kierron, jonka viimeksi asetit
-                yksilöllisesti. Tämä voi olla mukavampi vaihtoehto jos usein
-                sijoitat eri rakennustyyppejä.
+            title: Kiertäminen laitetyypin mukaan
+            description: Muistaa jokaisen laitetyypin viimeisimmän kiertoasetuksen.
+		Tämä voi olla mukavampi vaihtoehto jos sijoitat usein eri laitetyyppejä.
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:834]: Too many spaces in front This needs to be 4 Tabs (16 spaces)

File: translations/base-fi.yaml [hunk 4]
```diff
@@ -831,84 +817,72 @@ settings:
...
-            description: Aseta äänenvoimakkuus musiikille
+            description: Aseta musiikin äänenvoimakkuus
         lowQualityMapResources:
-            title: Low Quality Map Resources
-            description: Simplifies the rendering of resources on the map when zoomed in to
-                improve performance. It even looks cleaner, so be sure to try it
-                out!
+            title: Alhaisen tason resurssit
+            description: Yksinkertaistaa resurssikenttien ulkonäön suurennetussa näkymässä
+		suorituskyvyn parantamiseksi. Näyttää jopa siistimmältä, joten kannattaa kokeilla!
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:852]: Too many spaces in front This needs to be 4 Tabs (16 spaces)

File: translations/base-fi.yaml [hunk 5]
```diff
@@ -831,84 +812,72 @@ settings:
...
             description: Ottaa käyttöön vinjetoinnin, joka tummentaa näytön kulmia ja tekee
                 tekstin lukemisesta helpompaa.
         rotationByBuilding:
-            title: Kiertäminen rakennustyypin mukaan
-            description: Jokainen rakennustyyppi muistaa kierron, jonka viimeksi asetit
-                yksilöllisesti. Tämä voi olla mukavampi vaihtoehto jos usein
-                sijoitat eri rakennustyyppejä.
+            title: Kiertäminen laitetyypin mukaan
+            description: Muistaa jokaisen laitetyypin viimeisimmän kiertoasetuksen.
+		        Tämä voi olla mukavampi vaihtoehto jos sijoitat usein eri laitetyyppejä.
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:829]: Too many spaces in front. Should be 4 Tabs (16 spaces) in total

File: translations/base-fi.yaml [hunk 6]
```diff
@@ -831,84 +812,72 @@ settings:
...
-            description: Aseta äänenvoimakkuus musiikille
+            description: Aseta musiikin äänenvoimakkuus
         lowQualityMapResources:
-            title: Low Quality Map Resources
-            description: Simplifies the rendering of resources on the map when zoomed in to
-                improve performance. It even looks cleaner, so be sure to try it
-                out!
+            title: Alhaisen tason resurssit
+            description: Yksinkertaistaa resurssikenttien ulkonäön suurennetussa näkymässä
+		        suorituskyvyn parantamiseksi. Näyttää jopa siistimmältä, joten kannattaa kokeilla!
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:847]: Too many spaces in front. Should be 4 Tabs (16 spaces) in total

File: translations/base-fi.yaml [hunk 7]
```diff
@@ -513,92 +505,89 @@ buildings:
...
+            name: Virtuaalimaalain
+            description: Maalaa virtuaalisesti tavara alhaalta sisääntulosta oikean sisääntulon värillä.
     item_producer:
         default:
-            name: Item Producer
-            description: Saatavilla vain hiekkalaatikkotilassa. Palauttaa
-                johtotasolla annetun signaalin normaaliin tasoon.
+            name: Signaaligeneraattori
+            description: Saatavilla vain hiekkalaatikkotilassa. Lähettää
+            johtotasolla annetun signaalin normaaliin tasoon.
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:576]: Too few spaces in front (only 12). Should be 4 Tabs (16 spaces) in total

**Review by DerFeldspatz — CHANGES_REQUESTED** [review]

File: translations/base-fi.yaml [hunk 8]
```diff
@@ -662,82 +651,75 @@ storyRewards:
...
+            alue (pidä CTRL pohjassa ja raahaa hiirellä), ja paina 'C'
+            kopioidaksesi piirrustuksen.<br><br>Sen liittäminen ei ole
+            <strong>ilmaista</strong>. Sinun täytyy tuottaa <strong>piirustusmuotoja</strong>,
+            jotta sinulla on varaa siihen! (Ne mitkä juuri toimitit).
     no_reward:
         title: Seuraava taso
-        desc: "This level gave you no reward, but the next one will! <br><br> PS: Better
-            don't destroy your existing factory - You need <strong>all</strong>
-            those shapes later again to <strong>unlock upgrades</strong>!"
+        desc: Et saanut palkintoa tältä tasolta, mutta seuraavalta tasolta saat! <br><br> PS: Parempi
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:660]: Put everything after `desc: ` in quotation marks `"`

Comment (DerFeldspatz) [reply translations/base-fi.yaml:660]: The whole paragraph

File: translations/base-fi.yaml [hunk 9]
```diff
@@ -662,82 +651,75 @@ storyRewards:
...
+        desc: Avasit juuri 180-asteen <strong>Kääntäjän</strong>! - Sillä voit
+            kääntää muotoa 180 astetta (Ylläripylläri! :D)
     reward_display:
-        title: Display
-        desc: "You have unlocked the <strong>Display</strong> - Connect a signal on the
-            wires layer to visualize it!<br><br> PS: Did you notice the belt
-            reader and storage output their last read item? Try showing it on a
-            display!"
+        title: Näyttö
+        desc: Avasit juuri <strong>Näytön</strong> - Yhdistä signaali näyttöön
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:685]: Same here with the quotation marks

File: translations/base-fi.yaml (hunk 8 above)
Comment (DerFeldspatz) [reply translations/base-fi.yaml:660]: Like in line 317ff. `:` colon is a key that makes the lint fail if you dont quote the whole translation

File: translations/base-fi.yaml [hunk 10]
```diff
@@ -662,82 +651,75 @@ storyRewards:
...
+            <strong>muodon</strong>, <strong>värin</strong> tai <strong>binääriarvon</strong> (1 / 0).
     reward_logic_gates:
-        title: Logic Gates
-        desc: You unlocked <strong>logic gates</strong>! You don't have to be excited
-            about this, but it's actually super cool!<br><br> With those gates
-            you can now compute AND, OR, XOR and NOT operations.<br><br> As a
-            bonus on top I also just gave you a <strong>transistor</strong>!
+        title: Logiikkaportit
+        desc: Avasit <strong>Logiikkaportit</strong> (heh,heh)! Niistä ei tarvitse innostua,
+		    mutta ne ovat oikeasti tosi päheitä!<br><br> Logiikkaporteilla
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml]: Check your spaces in front here

File: translations/base-fi.yaml [hunk 11]
```diff
@@ -662,82 +651,75 @@ storyRewards:
...
+            voit suorittaa AND, OR, XOR ja NOT operaatioita.<br><br> Kaupanpäällisiksi sait
+		    myös <strong>transistorin</strong>!
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml]: and here

**Review by DerFeldspatz — CHANGES_REQUESTED** [review]

File: translations/base-fi.yaml [hunk 12]
```diff
@@ -934,65 +904,54 @@ keybindings:
...
-        lever: Switch (regular)
-        filter: Filter
-        wire_tunnel: Wire Crossing
-        display: Display
-        reader: Belt Reader
-        virtual_processor: Virtual Cutter
-        transistor: Transistor
-        analyzer: Shape Analyzer
-        comparator: Compare
-        item_producer: Item Producer (Sandbox)
```
Comment (DerFeldspatz) [inline translations/base-fi.yaml:980]: Why is this whole part missing???

**Review by DerFeldspatz — CHANGES_REQUESTED** [review]

File: translations/base-fi.yaml (hunk 12 above)
Comment (DerFeldspatz) [reply translations/base-fi.yaml:980]: Below it

File: translations/base-fi.yaml (hunk 8 above)
Comment (turbomursu) [reply translations/base-fi.yaml]: noticed that.

File: translations/base-fi.yaml (hunk 12 above)
Comment (turbomursu) [reply translations/base-fi.yaml:980]: I have no idea. Maybe I messed it up originally. It's now back there. Hope it helps with the errors too

Comment (turbomursu) [reply translations/base-fi.yaml:980]: YAY! It's error free if I read the log correctly. Finally I can go to sleep

**Comment by tobspr** [comment]
Merged in #918

**Pull request closed without merging**